scripts/syncup-s3.sh

```diff
@@ -0,0 +1,46 @@
+#! /bin/bash
+
+# This script is used to sync files required for training module notebooks to S3
+
+set -euo pipefail
+
+# Set the working directory to the directory of this file
+cd "$(dirname "${BASH_SOURCE[0]}")"
+# move back up to the training modules root
+cd ..
+
+bucket=s3://ccdl-training-data/training-modules
+
+sync_dirs=(
+  intro-to-R-tidyverse/data
+  RNA-seq/data/gastric-cancer/salmon_quant
+  RNA-seq/data/NB-cell/tximport
+  scRNA-seq/data/tabula-muris/alevin-quant/10X_P4_3
+  scRNA-seq/index/Mus_musculus
+  machine-learning/data/open-pbta/processed
+)
+
+sync_files=(
+  RNA-seq/data/gastric-cancer/gastric-cancer_metadata.tsv
+  RNA-seq/data/NB-cell/NB-cell_metadata.tsv
+  RNA-seq/index/Homo_sapiens/Homo_sapiens.GRCh38.95_tx2gene.tsv
+  scRNA-seq/data/glioblastoma/preprocessed/tximport/count_matrix.tsv
+  scRNA-seq/data/glioblastoma/preprocessed/darmanis_metadata.tsv
+  scRNA-seq/data/tabula-muris/normalized/TM_normalized.rds
+  pathway-analysis/data/medulloblastoma/medulloblastoma_vst_collapsed.tsv
+)
```

Comment:
It took me a second to wrap my head around why we're distinguishing between directories and files here, which suggests to me that a couple comments could be helpful!

Reply:
Yes, I should make this explicit. It was the result of some hard-won knowledge that `sync` doesn't work for individual files. (I already knew that `cp` had no recursive option.) 

I did debate at one point using some variant of `find -L` to get the file lists, but then I would lose the advantage of not recopying that `sync` gives. It is probably a small difference and wasn't worth worrying about, but I don't necessarily mind distinguishing directories from files?

Reply:
It is perfectly fine as is IMO, just want to make sure a little more info makes it into the comments!